### PR TITLE
Add barebones version of electric coil barrel and refactor barrel animation

### PR DESCRIPTION
**Pull request by toberge**

### CODE REVIEW

File: Assets/Scripts/Augment/AugmentImplementations/BarrelAnimator.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+using UnityEngine;
+
+/// <summary>
+/// Abstract for reload and fire animations.
+/// Not an interface since interface types can't be editor fields :(
```
Comment (Fueredoriku) [inline Assets/Scripts/Augment/AugmentImplementations/BarrelAnimator.cs:5]: Unity pain ☢

File: Assets/Scripts/Augment/BulletController.cs [hunk 2]
```diff
@@ -14,34 +16,62 @@ public class BulletController : ProjectileController
...
+
+        // TODO should be in projectilecontroller :)
+        gunController = transform.parent.GetComponent<GunController>();
+        if (!gunController)
+        {
+            Debug.Log("HatBarrel not attached to gun parent!");
+            return;
+        }
+        gunController.onInitializeGun += OnInitialize;
+        //gunController.onFire += OnFire;
```
Comment (Fueredoriku) [inline Assets/Scripts/Augment/BulletController.cs:47]: Why is onFire removed? 🤔

**Review by Fueredoriku — APPROVED** [review]
Looks great! 👍 

I only have a small comment, but I've tested and looked through the code, so I'll leave an approve for you 🤠

File: Assets/Scripts/Augment/BulletController.cs (hunk 2 above)
Comment (toberge) [reply Assets/Scripts/Augment/BulletController.cs:47]: It was replaced with a call to InitializeProjectile (and not invoked anymore), at this point we could actually subscribe to that delegate again though >_>
and I may have just copied this over from HatBarrel without putting too much thought into it
and these names give me pain btw :))))

Comment (toberge) [reply Assets/Scripts/Augment/BulletController.cs]: There, it's cleaned up now... Though please test the game, seems like I broke the hat barrel on my end :/

Comment (toberge) [reply Assets/Scripts/Augment/BulletController.cs]: ok thanks for fixing my mistake fredrik

**Pull request merged**